Some tests failing with moto 5.0.0 and pytest 8.0.0

**Describe the bug**

**moto-related errors**

The recent moto 5.0.0 release is causing several tests to fail in freshly installed environments (including CI) with

```text
E           Exception: Failed to start moto server after 60000 ms
```

This in turn is caused by an error in starting the moto server:

```text
test/s3test.py:43: Exception
------------------------------------------------------------------------------------------------------------ Captured stderr setup -------------------------------------------------------------------------------------------------------------
usage: server.py [-h] [-H HOST] [-p PORT] [-r] [-s] [-c SSL_CERT] [-k SSL_KEY]
server.py: error: unrecognized arguments: s3
_________________________________________________________________________________
```

The affected tests are:

```text
test/core/test_dsio.py::ZarrDatasetS3IOTest::test_write_to_and_read_from_s3
test/core/test_dsio.py::GetPathOrObsStoreTest::test_path_or_store_read_from_bucket
test/core/test_dsio.py::GetPathOrObsStoreTest::test_path_or_store_read_from_local
test/core/test_dsio.py::GetPathOrObsStoreTest::test_path_or_store_write_to_bucket
test/core/gridmapping/test_cfconv.py::AddSpatialRefTest::test_add_spatial_ref
test/core/store/fs/test_registry.py::S3FsDataStoresTest::test_dataset_levels
test/core/store/fs/test_registry.py::S3FsDataStoresTest::test_dataset_netcdf
test/core/store/fs/test_registry.py::S3FsDataStoresTest::test_dataset_zarr
test/core/store/fs/test_registry.py::S3FsDataStoresTest::test_mldataset_levels
test/core/store/fs/impl/test_geotiff.py::ObjectStorageMultiLevelDatasetTest::test_s3_fs
test/core/zarrstore/test_generic.py::CommonS3ZarrStoreTest::test_it
```

**pytest-related errors**

```text
FAILED test/cli/test_main.py::ClickMainTest::test_warnings_issued_when_enabled - pytest.PytestRemovedIn8Warning: Passing None has been deprecated.
FAILED test/cli/test_main.py::ClickMainTest::test_warnings_not_issued_by_default - pytest.PytestRemovedIn8Warning: Passing None has been deprecated.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Check out current repo (98f6a37aa26dcf)
2. Create a fresh environment from the environment.yml and install xcube from the repo.
3. Run the test suite.
4. Note failures.

**Expected behavior**
All tests should pass.

**Discussion**

**moto-related errors**

As the error message says, the argument `s3` to moto's `server.py` is no longer recognized. The change is mentioned in <https://github.com/getmoto/moto/pull/7220>:

> Remove `moto_server` `service` argument - we can just run `moto_server` for all services, no need to run a separate server per service anymore

So the solution should be simply to omit the `s3` argument when the server is started from `s3test.py`.

**pytest-related errors**

As the error message indicates, `None` is no longer a valid argument to `pytest.warns` . In both the tests listed above, `pytest.warns` was being used to catch emitted warnings for checking. The `None` can be replaced with `Warning` or with a tuple of more specific expected warning types.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -28,6 +28,7 @@
 ### Fixes
 
 * Make S3 unit tests compatible with moto 5 server (#922)
+* Make some CLI unit tests compatible with pytest 8 (#922)
 
 ### Other changes
 

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -25,6 +25,10 @@
   * Set Content-Bbox and Content-Crs headers in the HTTP response
   * Support safe CURIE syntax for CRS specification
 
+### Fixes
+
+* Make S3 unit tests compatible with moto 5 server (#922)
+
 ### Other changes
 
 * Require Python >=3.9 (previously >=3.8)

--- a/test/cli/test_main.py
+++ b/test/cli/test_main.py
@@ -16,19 +16,25 @@ class ClickMainTest(CliTest):
         @command(name='test_warnings')
         def test_warnings():
             warnings.warn(self.USER_WARNING_MESSAGE)
-            warnings.warn(self.DEPRECATION_WARNING_MESSAGE, category=DeprecationWarning)
-            warnings.warn(self.RUNTIME_WARNING_MESSAGE, category=RuntimeWarning)
+            warnings.warn(
+                self.DEPRECATION_WARNING_MESSAGE, category=DeprecationWarning
+            )
+            warnings.warn(
+                self.RUNTIME_WARNING_MESSAGE, category=RuntimeWarning
+            )
 
         xcube.cli.main.cli.add_command(test_warnings)
 
     def test_warnings_not_issued_by_default(self):
-        with pytest.warns(None) as record:
+        with pytest.warns(UserWarning) as record:
             self.invoke_cli(['test_warnings'])
         self.assertEqual(['This is a user warning.'],
                          list(map(lambda r: str(r.message), record.list)))
 
     def test_warnings_issued_when_enabled(self):
-        with pytest.warns(None) as record:
+        with pytest.warns(
+            (UserWarning, DeprecationWarning, RuntimeWarning)
+        ) as record:
             self.invoke_cli(['--warnings', 'test_warnings'])
         self.assertEqual(['This is a user warning.',
                           'This is a deprecation warning.',

--- a/test/s3test.py
+++ b/test/s3test.py
@@ -12,7 +12,7 @@ import moto.server
 MOTO_SERVER_ENDPOINT_URL = f'http://127.0.0.1:5000'
 
 MOTOSERVER_PATH = moto.server.__file__
-MOTOSERVER_ARGS = [sys.executable, MOTOSERVER_PATH, 's3']
+MOTOSERVER_ARGS = [sys.executable, MOTOSERVER_PATH]
 
 
 class S3Test(unittest.TestCase):

